BigInt.c

```diff
@@ -119,6 +119,42 @@ BigInt* BigInt_construct(int value) {
     return new_big_int;
 }
 
+BigInt* BigInt_from_string(const char* str) {
+    BOOL is_negative = (*str == '-');
+    if( is_negative ){
+        str++;
+    }
+    while( *str == '0' ){ // remove leading zeros
```

Comment:
If the input string is just "0", "-0", or "0000" etc we'll go too far here.

Would also be good to have a test case for that.